### ISSUE TITLE
Installation fails due to cmake error detecting functional CXX compiler

### ISSUE THREAD

**Issue by wresch** [SimpleITK/SimpleITKRInstaller#33]
R 3.6.0
cmake 3.12.1
gcc 7.3.0

doing a basic `devtools::install_github("SimpleITK/SimpleITKRInstaller", args=c('--configure-vars="MAKEJ=6"'))` fails with

```
> devtools::install_github("SimpleITK/SimpleITKRInstaller",
+     args=c("--configure-vars=\"MAKEJ=6\""))
Downloading GitHub repo SimpleITK/SimpleITKRInstaller@master
   checking for file ‘/tmp/RtmpNuxMMR/remotesd3fc5cdce881/SimpleITK-SimpleITKRInstaller-bc83a93/DESCRIPTION’
✔  checking for file ‘/tmp/RtmpNuxMMR/remotesd3fc5cdce881/SimpleITK-SimpleITKRInstaller-bc83a93/DESCRIPTION’
─  preparing ‘SimpleITK’:
✔  checking DESCRIPTION meta-information ...
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
   Removed empty directory ‘SimpleITK/.circleci’
─  creating default NAMESPACE file
   Warning: invalid gid value replaced by that for user 'nobody'
─  building ‘SimpleITK_1.2.0-19.tar.gz’

Installing package into ‘/usr/local/apps/R/3.6/site-library_3.6.0’
(as ‘lib’ is unspecified)
* installing *source* package ‘SimpleITK’ ...
** using staged installation
/tmp/RtmpJtPmK2/R.INSTALLd4252b761ab/SimpleITK
Cloning into 'SimpleITK'...
remote: Enumerating objects: 62, done.
remote: Counting objects: 100% (62/62), done.
remote: Compressing objects: 100% (45/45), done.
remote: Total 42333 (delta 22), reused 38 (delta 12), pack-reused 42271
Receiving objects: 100% (42333/42333), 18.61 MiB | 30.13 MiB/s, done.
Resolving deltas: 100% (24505/24505), done.
Note: checking out '7c84168ffecbdcfafdc3c9faaf6331d6684dc24c'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b new_branch_name

HEAD is now at 7c84168... Merge branch 'FixRManPath' into release
-- The C compiler identification is GNU 7.3.0
-- The CXX compiler identification is GNU 7.3.0
-- Check for working C compiler: /usr/local/GCC/7.3.0/bin/gcc
-- Check for working C compiler: /usr/local/GCC/7.3.0/bin/gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/local/GCC/7.3.0/bin/g++
-- Check for working CXX compiler: /usr/local/GCC/7.3.0/bin/g++ -- broken
CMake Error at /usr/local/Cmake/3.12.1/share/cmake-3.12/Modules/CMakeTestCXXCompiler.cmake:45 (message):
  The C++ compiler

    "/usr/local/GCC/7.3.0/bin/g++"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: /tmp/RtmpJtPmK2/R.INSTALLd4252b761ab/SimpleITK/SITK/Build/CMakeFiles/CMakeTmp

    Run Build Command:"/bin/gmake" "cmTC_e9453/fast"
    make -f CMakeFiles/cmTC_e9453.dir/build.make CMakeFiles/cmTC_e9453.dir/build
    make[1]: Entering directory `/tmp/RtmpJtPmK2/R.INSTALLd4252b761ab/SimpleITK/SITK/Build/CMakeFiles/CMakeTm
p'
    Building CXX object CMakeFiles/cmTC_e9453.dir/testCXXCompiler.cxx.o
    /usr/local/GCC/7.3.0/bin/g++  -std=gnu++11    ERROR: no information for variable 'CXX1XFLAGS'    -o CMake
Files/cmTC_e9453.dir/testCXXCompiler.cxx.o -c /tmp/RtmpJtPmK2/R.INSTALLd4252b761ab/SimpleITK/SITK/Build/CMake
Files/CMakeTmp/testCXXCompiler.cxx
    g++: error: ERROR:: No such file or directory
    g++: error: no: No such file or directory
    g++: error: information: No such file or directory
    g++: error: for: No such file or directory
    g++: error: variable: No such file or directory
    g++: error: CXX1XFLAGS: No such file or directory
    make[1]: *** [CMakeFiles/cmTC_e9453.dir/testCXXCompiler.cxx.o] Error 1
    make[1]: Leaving directory `/tmp/RtmpJtPmK2/R.INSTALLd4252b761ab/SimpleITK/SITK/Build/CMakeFiles/CMakeTmp
'
    gmake: *** [cmTC_e9453/fast] Error 2




  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:16 (project)


-- Configuring incomplete, errors occurred!
See also "/tmp/RtmpJtPmK2/R.INSTALLd4252b761ab/SimpleITK/SITK/Build/CMakeFiles/CMakeOutput.log".
See also "/tmp/RtmpJtPmK2/R.INSTALLd4252b761ab/SimpleITK/SITK/Build/CMakeFiles/CMakeError.log".
ERROR: configuration failed for package ‘SimpleITK’
* removing ‘/usr/local/apps/R/3.6/site-library_3.6.0/SimpleITK’
Error in i.p(...) :
  (converted from warning) installation of package ‘/tmp/RtmpNuxMMR/filed3fc69832e53/SimpleITK_1.2.0-19.tar.gz’ had non-zero exit status
```

I suppose this may be a SimpleITK issue more than a SimpleITKRInstaller issue. Anything obvious jump out?

**Comment by richardbeare** [comment]
The most recent version of R have switched the various CXX flags over from CXX1X to CXX11. We've been a bit slack about pushing this through.

There's a fixed branch that might work for you:

```r
devtools::install_github("richardbeare/SimpleITKRInstaller", ref="TravisTests", args=c('--configure-vars="MAKEJ=6"'))
```

**Comment by wresch** [comment]
Super - thank you. That worked (or at least it got me past this particular issue - compilation just started and I know this takes a while). And it seems to be the same version of SimpleITK.

**Comment by wresch** [comment]
and compilation finished successfully. Thanks again.

**Comment by blowekamp** [comment]
@richardbeare Can you please make a PR with this fix. Thanks!

**Comment by blowekamp** [comment]
closed by #35
